### ISSUE TITLE
🐛 `useIframeSandbox` doesn't detect constants

### ISSUE THREAD

**Issue by jakeleventhal** [biomejs/biome#10032]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.12
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.12.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.12

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

</details>

### What happened?

https://github.com/jakeleventhal/biome-use-iframe-sandbox-constant-repro

### Expected result

No suppression

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
